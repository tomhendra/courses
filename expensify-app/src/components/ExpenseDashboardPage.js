import React from 'react';

const ExpenseDashboardPage = () => (
  <div>
    This is from the dashboard component
  </div>
);

export default ExpenseDashboardPage;