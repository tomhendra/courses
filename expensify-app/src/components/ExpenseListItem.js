import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <h3>{description}</h3>
    <p>Amount: &euro;{amount} | Date created: {createdAt}</p>
    <button onClick={() => dispatch(removeExpense({id}))}>Remove</button>
  </div>
);

export default connect()(ExpenseListItem);
