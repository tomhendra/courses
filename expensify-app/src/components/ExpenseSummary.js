import React from 'react';
import { connect } from 'react-redux';
import numberFormatter from '../utils/number-formatter';
import selectExpenses from './../selectors/expenses';
import selectExpensesTotal from './../selectors/expenses-total';

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' ;
  const formattedExpensesTotal = numberFormatter.format(expensesTotal / 100);
  
  return (
    <div>
      <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
