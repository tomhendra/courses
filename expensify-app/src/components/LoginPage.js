import React from 'react';

export const LoginPage = () => (
  <div>
    <form>
      <button>Login</button>
    </form>
  </div>
);
