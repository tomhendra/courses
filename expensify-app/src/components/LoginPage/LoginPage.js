import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../../redux/actions/auth';

import style from './LoginPage.scss';

export const LoginPage = ({ startLogin }) => (
  <div className={style['box-layout']}>
    <button onClick={startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage);
