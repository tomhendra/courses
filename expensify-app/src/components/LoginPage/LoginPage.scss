.box-layout {
  background: url('../../assets/images/bg.jpg');
  background-color: cadetblue;
}