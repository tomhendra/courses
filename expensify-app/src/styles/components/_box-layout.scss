div {
  background:url('/src/images/bg.jpg');
}
