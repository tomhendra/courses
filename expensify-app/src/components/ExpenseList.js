import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from './../selectors/expenses';

// regular unconnected component
export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
        props.expenses.map(expense => <ExpenseListItem {...expense} key={expense.id} />)
      )
    }
    {/* destructuring props from object using {...expense} */}
  </div>
);

// functions to define what we want to pull from the store
const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

// call to connect() to pull everything together
export default connect(mapStateToProps)(ExpenseList);

// when you connect a component to the store, it becomes reactive -- as the store changes, the component will be re-rendered with the new values from the store
