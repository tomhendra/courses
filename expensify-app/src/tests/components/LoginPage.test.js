import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('Should render login page correctly', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});
