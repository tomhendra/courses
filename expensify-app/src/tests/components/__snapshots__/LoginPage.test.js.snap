// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render login page correctly 1`] = `
<div>
  <form>
    <button>
      Login
    </button>
  </form>
</div>
`;
